src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -0,0 +1,720 @@
+                __export_keyvalue(key_segments,
+                                  kv.value, exported_dict, key)
+
+        if exported_dict and exported_list:
+            logger.error("Can not export to a valid file! Some keys have been dropped. %s", json.dumps(
+                exported_dict, indent=2, ensure_ascii=False))
+
+        return __compact_key_values(exported_dict if not exported_list else exported_list)
+    except Exception as exception:
+        raise CLIError("Fail to export key-values." + str(exception))
+
+
+def __flatten_dict_to_properties(value_to_flatten, separator='.', prefix=''):
+    flattened_properties_dict = {}
+    if isinstance(value_to_flatten, dict):
+        for keys, values in value_to_flatten.items():
+            current_prefix = prefix + separator + str(keys) if prefix else str(keys)
+            if isinstance(values, (dict, list)):
+                flattened_properties_dict.update(__flatten_dict_to_properties(values, separator, current_prefix))
+            else:
+                flattened_properties_dict.update({current_prefix: str(values)})
+
+    elif isinstance(value_to_flatten, list):
+        for idx, val in enumerate(value_to_flatten):
+            current_prefix = prefix + '[' + str(idx) + ']' if prefix else str(idx)
+            if isinstance(val, (dict, list)):
+                flattened_properties_dict.update(__flatten_dict_to_properties(val, separator, current_prefix))
+            else:
+                flattened_properties_dict.update({current_prefix: str(val)})
+    else:
+        flattened_properties_dict.update({prefix: str(value_to_flatten)})
+
+    return flattened_properties_dict
+
+
+def __export_features(retrieved_features, format_):
+    exported_dict = {}
+    client_filters = []
+    if format_ == 'json':
+        exported_dict["FeatureManagement"] = {}
+
+    elif format_ in ('yaml', 'properties'):
+        # Currently, this condition will never be met
+        # We only support json feature flags for now
+        featureSetValues = {}
+        featureSetValues["features"] = {}
+        exported_dict["feature-management"] = {"featureSet": featureSetValues}
+
+    try:
+        # retrieved_features is a list of FeatureFlag objects
+        for feature in retrieved_features:
+
+            # if feature state is on or off, it means there are no filters
+            if feature.state == "on":
+                feature_state = True
+
+            elif feature.state == "off":
+                feature_state = False
+
+            elif feature.state == "conditional":
+                feature_state = {}
+                feature_state["EnabledFor"] = []
+                client_filters = feature.conditions["client_filters"]
+                # client_filters is a list of dictionaries, where all dictionaries have 2 keys - Name and Parameters
+                for filter_ in client_filters:
+                    feature_filter = {}
+                    feature_filter["Name"] = filter_.name
```

Comment:
>feature_filter [](start = 20, length = 14)

shouldn't this be FeatureFilter type?

Reply:
We want to keep it consistent with the .NET schema for JSON files, so the attribute has to be "Name" instead of "name" from FeatureFilter object. Same for parameters.

---
In reply to: [356333207](https://github.com/Azure/azure-cli/pull/11467#discussion_r356333207) [](ancestors = 356333207)

Reply:
The retrieved_features was originally got from list_feature() which will return a list of FeatureFlag, then featureFlag.condition["client_filters"] should be a list of FeatureFilter objects instead of dictionary. 

---
In reply to: [356347693](https://github.com/Azure/azure-cli/pull/11467#discussion_r356347693) [](ancestors = 356347693,356333207)